### ISSUE TITLE
Jumping to Home Menu instead of mii maker

### ISSUE THREAD

**Issue by naendow** [dimok789/homebrew_launcher#19]
Hey

I know that it is not a problem, but homebrew launcher would feel much more like a real app if it would jump to home menu instead of mii maker.
Other apps like NNU-Patcher can jump to it, too. So I hope this is possible.

Oh, and controlling with Pro Controller would be great :D

**Comment by CreeperMario** [comment]
Hmm... I'll look into this later on if @dimok789 doesn't.
About the Wii U Pro Controller support, well, that's already half implemented. Using the L and R buttons will change screens.

**Comment by dimok789** [comment]
Actually i did that on purpose as you would loose all possibility to enter mii maker otherwise.

**Comment by naendow** [comment]
So there is no way to get into the home menu like i.e. NNU-Patcher does?

**Comment by dimok789** [comment]
There is a way. I ommited it on purpose is what I am mean.

**Comment by Calahan99** [comment]
Maybe it is possible to create a menu item in Homebrew Launcher named "Mii Maker" to start/go to Mii Maker, so Homebrew Launcher could exit to home menu as default?

**Comment by CreeperMario** [comment]
@Calahan99 the HOME Menu on a Wii U console is controlled by a library called ProcUI.
Nobody has yet been able to load ProcUI into an ELF-based program, though that doesn't mean it is impossible. But don't be surprised if it doesn't work.

**Comment by Leseratte10** [comment]
Push again. I'd rather have the HBL exit to the System menu when I press the home button than to the Mii Maker. 

Maybe you can just add a little button in the corner to boot the Mii Maker? Or a menu (like the HBC on Wii) which you can open with "+" or "-" and which shows options like to boot the Mii Maker?

**Comment by CreeperMario** [comment]
Resolved in my homebrew_launcher fork.
When you press HOME to exit, Mii Maker will launch.
If you hold down SELECT (the minus button) while pressing HOME, the Wii U Menu will launch instead.

Will PR soon, got more issues to fix. 😉

**Comment by SirNapkin1334** [comment]
Take all the time you need. I'm sifting through the files right now, trying to aid my c learning by making sense of it.

**Comment by CreeperMario** [comment]
See #39

When the HOME button is pressed down (which is checked comparing the bit-mask `controller->data.buttons_d` with the constant value `VPAD_BUTTON_MINUS`, which detects which buttons have just been pressed down), usually, it would simply change the boolean value `exitApplication`, flagging to the graphics thread that the program should start de-initialising itself.

I have added a check here that detects if the SELECT button on the controller has been held down (using `controller->data.buttons_h`, detects which buttons have been held for more than one loop iteration) and, if so, `SYSLaunchMenu` is called. This function flags to the PPC kernel that once the program has finished, the Wii U Menu should open. This overrides the behaviour provided by `sd_loader` which requests that, once the program has finished, that Mii Maker should be re-launched.

If you need any more help, let me know, I'm happy to help.

**Comment by SirNapkin1334** [comment]
Why leave the ability to exit to Mii maker in at all?

**Comment by SirNapkin1334** [comment]
Alright, I (hopefully) did what I wanted which was reverse the inputs, but upon making the files, I'm presented with an error that probably doesn't relate to this issue at all, but I might as well tell you just in case:
![screen shot 2017-10-13 at 20 14 55](https://user-images.githubusercontent.com/31591796/31572134-be651304-b053-11e7-8def-69a46b404a0d.png)

**Comment by CreeperMario** [comment]
You'll need to install @dimok789's modified versions of [libogc](https://github.com/dimok789/homebrew_launcher/releases/download/v1.3/libogc.zip) and the [PowerPC portlibs](https://github.com/dimok789/homebrew_launcher/releases/download/v1.3/portlibs.zip). You just need to put the folders from these archives into the `$DEVKITPRO` folder.

**Comment by SirNapkin1334** [comment]
I did that, but now, we have some problem with you're script. I'm smart enough in C to know that you haven't declared the function `SYSLauncMenu()` , and it gives me an error:
![screen shot 2017-10-14 at 07 08 16](https://user-images.githubusercontent.com/31591796/31576388-dbc8063a-b0ae-11e7-96bd-dd5564d8c724.png)
Also, there are some other, apparently less important, errors above. The only changes I made to your code was to switch around the inputs for exit and menu:
![screen shot 2017-10-14 at 07 04 16](https://user-images.githubusercontent.com/31591796/31576411-05dc59d0-b0af-11e7-8fb5-de152f425319.png)
However, I was wondering why in your code you hadn't declared `SYSLaunchMenu()` , and figured if I used your code it might not work for the same reason. Any help on where to declare and what to put under the function?

**Comment by SirNapkin1334** [comment]
Alright, I directly copied and pasted your code in, did a little switcheroo, and now the function isn't an error anymore, except that the errors that were above that error are now halting the compilation. Any help (again)?

**Comment by CreeperMario** [comment]
`SYSLaunchMenu` is defined in `dynamic_libs/sys_functions.h`. You'll also notice in `main.cpp` that I have run `InitSysFunctionPointers`, which initialises the `SYS` functions.

Those errors about `int`s and `s32`s are normal. Unfortunately, there were some upstream changes in GCC that were merged into devkitPPC r28 and r29, which cause issues with certain software. For now, you must use [devkitPPC r27](https://sourceforge.net/projects/devkitpro/files/devkitPPC/) to build the Homebrew Launcher, and all software based on @dimok789's graphics library.

**Comment by SirNapkin1334** [comment]
Okay. I see. Thanks.

**Comment by SirNapkin1334** [comment]
Okay, it built fine, but now, well, it RIP's when loading from CBHC, which defeats the entire purpose. It crashes while booting the HBL. Loading from the Wii U menu is fine, as I had expected. Any fix?

**Comment by CreeperMario** [comment]
Not sure, I've never used CBHC.

What do you mean by "RIP?" Does it display a blank screen? What do you hear? Any buzzing? Do you have to hold the power button on your console to shut it down?

**Comment by SirNapkin1334** [comment]
Yeah, sorry, I put it really vaguely. I didn't brick my console, don't worry. Upon booting into HBL from CBHC, it shows that white gradient screen with the Wii U logo in the bottom right corner. The Wii U boot music loops a few times, and then it crashes, still displaying the Wii U screen, and I hear buzzing. The buzzing seemed oddly more quiet than would be expected, from past crashes. I'm forced to hold down the Wii U power button. I'd send a video, but it'd be way to large for GitHub.
